### PR TITLE
mgr/dashboard_v2: Add iSCSI page

### PR DESCRIPTION
**Pull request by ricardoasmarques**
This PR adds the **frontend** part of the iSCSI page, and should be merged after https://github.com/openattic/ceph/pull/82 that includes the **backend** part.

screenshot from 2018-02-16 11-50-49

Signed-off-by: Ricardo Marques <rimarques@suse.com>

### CODE REVIEW

**Review by LenzGr — DISMISSED** [review]
LGTM. I like the usage of `moment` here!

**Comment by sebastian-philipp** [comment]
#82 is merged.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+import { TcmuIscsiService } from '../../../shared/services/tcmu-iscsi.service';
+
+@Component({
+  selector: 'cd-iscsi',
+  templateUrl: './iscsi.component.html',
+  styleUrls: ['./iscsi.component.scss']
+})
+export class IscsiComponent implements OnInit {
+
+  daemons: Array<any> = null;
```
Comment (votdev) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts]: Doesn't ``daemons: Array<any> = [];`` make more sense here?

File: src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+  daemonsColumns: any;
+  images: Array<any> = null;
```
Comment (votdev) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts]: I suggest to use ``images: Array<any> = [];`` here.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  imagesColumns: any;
+  interval: any;
+
+  constructor(private tcmuIscsiService: TcmuIscsiService,
+              cephShortVersion: CephShortVersionPipe,
```
Comment (votdev) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts]: In most cases we use variable names like xxxYyyZzzPipe.

**Review by votdev — CHANGES_REQUESTED** [review]

File: src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts [hunk 4]
```diff
@@ -0,0 +1,110 @@
+import { Component, OnInit } from '@angular/core';
+import { CephShortVersionPipe } from '../../../shared/pipes/ceph-short-version.pipe';
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts]: Insert a blank line between lines 1 and 2.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts (hunk 1 above)
Comment (tspmelo) [reply src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts]: I also prefer to have an empty list, this way the table can be initialized properly.

`daemons = [];` will do the same.

File: src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+        pipe: relativeDate
+      },
+    ];
+
+  }
+
+  ngOnInit() {
+    this.refresh();
+
+    this.interval = setInterval(() => {
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard_v2/frontend/src/app/ceph/block/iscsi/iscsi.component.ts:99]: Don't forget to clear this interval when the component is destroyed.

**Review by tspmelo — CHANGES_REQUESTED** [review]

**Comment by ricardoasmarques** [comment]
@votdev @tspmelo all your comments were addressed. Please review again. Thanks.

**Review by votdev — APPROVED** [review]

**Review by LenzGr — APPROVED** [review]
LGTM!

**Review by tspmelo — APPROVED** [review]

**Pull request merged**